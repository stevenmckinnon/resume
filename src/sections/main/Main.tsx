import * as React from "react"
import { Link, Element } from "react-scroll"

import { type SocialMedia } from "../../interfaces"
import "./main.scss"
import {
  ChevronDown,
  Camera,
  Github,
  Instagram,
  Linkedin,
  Twitter,
} from "lucide-react"

interface Props {
  name: string
  intro: string
  socialMedia: SocialMedia
}

const Main: React.FC<Props> = ({ name, intro, socialMedia }) => {
  return (
    <Element
      name="home"
      className="home h-full flex flex-col align justify-center items-center p-6"
    >
      <div className="gap-3 flex flex-col align justify-center items-center max-w-4xl">
        <h1 className="text-white text-4xl md:text-8xl drop-shadow-md font-heading">{`I'm ${name}.`}</h1>

        <h3 className="text-white text-md md:text-lg font-body">
          {intro}
          <br />
          {`Let's `}
          <Link
            smooth
            className="smoothscroll text-white hover:text-teal-300 cursor-pointer"
            to="about"
          >
            start scrolling
          </Link>{" "}
          and learn more{" "}
          <Link
            smooth
            className="smoothscroll text-white hover:text-teal-300 cursor-pointer"
            to="about"
          >
            about me
          </Link>
          .
        </h3>
        <hr />
        <ul className="social flex gap-5 justify-center">
          {socialMedia.twitter != null && (
            <li>
              <a
                href={socialMedia.twitter}
                className="text-xl text-white hover:text-teal-300 cursor-pointer"
              >
                <Twitter />
              </a>
            </li>
          )}
          {socialMedia.linkedIn != null && (
            <li>
              <a
                href={socialMedia.linkedIn}
                className="text-xl text-white hover:text-teal-300 cursor-pointer"
              >
                <Linkedin />
              </a>
            </li>
          )}
          {socialMedia.instagram != null && (
            <li>
              <a
                href={socialMedia.instagram}
                className="text-xl text-white hover:text-teal-300 cursor-pointer"
              >
                <Instagram />
              </a>
            </li>
          )}
          {socialMedia.photography != null && (
            <li>
              <a
                href={socialMedia.photography}
                title={`${name} Photography`}
                className="text-xl text-white hover:text-teal-300 cursor-pointer"
              >
                <Camera />
              </a>
            </li>
          )}
          {socialMedia.github != null && (
            <li>
              <a
                href={socialMedia.github}
                className="text-xl text-white hover:text-teal-300 cursor-pointer"
              >
                <Github />
              </a>
            </li>
          )}
        </ul>
      </div>
      <div className="scrolldown absolute bottom-5 flex justify-center">
        <Link
          className="motion-safe:animate-bounce text-2xl text-white hover:text-teal-300 cursor-pointer"
          to="about"
          smooth={true}
        >
          <ChevronDown size={50} />
        </Link>
      </div>
    </Element>
  )
}

export default Main
