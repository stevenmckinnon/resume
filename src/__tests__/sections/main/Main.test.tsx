import { shallow } from "enzyme"
import * as React from "react"
import Main from "../../../sections/main/Main"

describe("Tests the Header component", () => {
  it("renders the Header component", () => {
    const name = "Steve McKinnon"
    const intro = `I'm a Glasgow based front end developer and web designer creating modern web apps in the financial sector.`
    const socialMedia = {
      twitter: "http://twitter.com/stevenmckinnon",
      linkedIn: "https://www.linkedin.com/in/stevenmckinnon92/",
      instagram: "http://instagram.com/stevenmckinnon",
      photography: "http://stevenmckinnon.co.uk",
      github: "https://github.com/stevenmckinnon",
    }
    const wrapper = shallow(
      <Main name={name} intro={intro} socialMedia={socialMedia} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it("renders the Header component", () => {
    const name = "Steve McKinnon"
    const socialMedia = {}
    const intro = `I'm a Glasgow based front end developer and web designer creating modern web apps in the financial sector.`
    const wrapper = shallow(
      <Main name={name} intro={intro} socialMedia={socialMedia} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
