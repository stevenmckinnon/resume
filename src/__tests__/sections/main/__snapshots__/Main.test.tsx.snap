// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Header component renders the Header component 1`] = `ShallowWrapper {}`;

exports[`Tests the Header component renders the Header component 2`] = `ShallowWrapper {}`;
